frontend/src/views/funding/FundingOverviewView.vue

```diff
@@ -54,47 +163,211 @@
+            await this.enrichFundingAgreementData(payload)
+            this.fundingAgreements.push(payload)
+          } catch (error) {
+            console.error(`Error getting funding agreements for facility ID ${facility.facilityId}:`, error)
+          }
+        }
+      } catch (error) {
+        console.error(`Error getting funding agreements:`, error)
+      } finally {
+        this.loading = false
+      }
+    },
+
+    async enrichFundingAgreementData(result) {
+      result.fundingAgreementType = 'Base Funding'
+      result.facility = this.facilityName(result.facilityId)
+      result.status = this.statusName(result.statusCode)
+      result.expenseAuthorityName = await this.getExpenseAuthorityName(result.applicationId)
+    },
+
+    async getExpenseAuthorityName(applicationId) {
+      try {
+        const application = await ApplicationService.getApplication(applicationId)
+        if (!application?.expenseAuthorityId) return ''
+        const expenseAuthorityUser = this.orgUsers.find(user => user.contactId === application.expenseAuthorityId)
+        return `${expenseAuthorityUser?.firstName} ${expenseAuthorityUser?.lastName}`
+      } catch (error) {
+        console.error('Error getting expense authority name:', error)
+        return ''
+      }
+    },
+
+    updateSelectedDateFilterType(value) {
+      this.selectedDateFilterType = value
+    },
+
+    startDateThreshold() {
+      switch (this.selectedDateFilterType) {
+        case '3Months':
+          return this.dateByMonthsInPast(3)
+        case '6Months':
+          return this.dateByMonthsInPast(6)
+        case 'YTD':
+          return this.dateByMonthsInPast(12)
+        default:
+          return null
+      }
+    },
+
+    dateByMonthsInPast(months) {
+      const date = new Date()
+      date.setMonth(date.getMonth() - months)
+      return date.toISOString().split('T')[0]
+    },
+
+    facilityName(facilityId) {
+      return this.userInfo.facilities.find(facility => facility.facilityId === facilityId).facilityName
+    },
+
+    statusName(statusCode) {
+      switch (statusCode) {
+        case FUNDING_AGREEMENT_STATUS_CODES.DRAFT:
+          return 'Draft'
+        case FUNDING_AGREEMENT_STATUS_CODES.FA_REVIEW:
+          return 'FA Review'
+        case FUNDING_AGREEMENT_STATUS_CODES.SIGNATURE_PENDING:
+          return 'FA Signature Pending'
+        case FUNDING_AGREEMENT_STATUS_CODES.SUBMITTED:
+          return 'FA Submitted to Ministry'
+        case FUNDING_AGREEMENT_STATUS_CODES.FA_IN_REVIEW:
+          return 'In Review with Ministry EA'
+        case FUNDING_AGREEMENT_STATUS_CODES.ACTIVE:
+          return 'Active'
+        case FUNDING_AGREEMENT_STATUS_CODES.EXPIRED:
+          return 'Expired'
+        default:
+          console.log('Unknown status code:', statusCode)
+          return 'Unknown'
+      }
+    },
+
+    getStatusClass(statusCode) {
+      if (this.FUNDING_AGREEMENT_STATUS_CODES.DRAFT === statusCode) {
```

Comment:
You can simplify this code greatly using object binding: https://vuejs.org/guide/essentials/class-and-style.

Reply:
I believe i got this right.

Reply:
You can use a method to return the object so that you don't inline complex logic.

Reply:
Fixed.

Reply:
Much simpler!